### PR TITLE
[5.5] Return request data from controller validate() call

### PR DESCRIPTION
**Pull request by JosephSilber**
So that instead of this:

```php
public function store()
{
    $this->validate(request(), [
        'title' => 'required',
        'body' => 'required',
    ]);

    $data = request()->only('title', 'body');

    Post::create($data);
}
```

You can now do this:

```php
public function store()
{
    $data = $this->validate(request(), [
        'title' => 'required',
        'body' => 'required',
    ]);

    Post::create($data);
}
```

### CODE REVIEW

**Comment by taylorotwell** [comment]
How does this behave with deeply nested "dot" notation validation rules?

**Comment by JosephSilber** [comment]
@taylorotwell do you mean regular dot notation, like this?

```php
$this->validate(request(), [
    'title' => 'required',
    'author.name' => 'required',
    'author.description' => 'required',
]);
```

This will end up calling `only` with the keys:

```php
$request->only('title', 'author.name', 'author.description');
```

Since `only` returns nested values, this will return the following:

```php
[
    'title' => 'The title',
    'author' => [
        'name' => 'The author's name',
        'description' => 'A description of the author',
    ],
]
```

**Comment by morloderex** [comment]
@JosephSilber Addition to @taylorotwell question. What would happen if one these keys is an uploaded file?

**Comment by JosephSilber** [comment]
@morloderex then you'd get the file as part of the returned array, same as calling `$request->only()`.

File: src/Illuminate/Foundation/Validation/ValidatesRequests.php [hunk 1]
```diff
@@ -23,11 +23,11 @@
      *
      * @param  \Illuminate\Contracts\Validation\Validator|array  $validator
      * @param  \Illuminate\Http\Request|null  $request
-     * @return void
+     * @return array
      */
     public function validateWith($validator, Request $request = null)
     {
-        $request = $request ?: app('request');
+        $request = $request ?: request();
```
Comment (laurencei) [inline src/Illuminate/Foundation/Validation/ValidatesRequests.php]: Why did this line need to change? I'm trying to understand the difference before/after?

**Comment by laurencei** [comment]
Does this have any effect when using FormRequest?

Could/should the `FormRequest` also only pass validated variables into the controller after validation? Could be an option to only allow validated data through?

**Comment by taylorotwell** [comment]
Seems like if we wanted a `validated` method could be added to FormRequest that does the same thing.

**Comment by taylorotwell** [comment]
Seems fine to me.

File: src/Illuminate/Foundation/Validation/ValidatesRequests.php (hunk 1 above)
Comment (JosephSilber) [inline src/Illuminate/Foundation/Validation/ValidatesRequests.php]: @laurencei correct.

**Pull request merged**